### PR TITLE
[CES-155] Updated static_analysis.yaml with new action and pre-commit updated

### PR DESCRIPTION
**Pull request by mamu0**
### Motivation and Context

The static_analysis.yaml action is changed to avoid docker use and update the action with more recent form.
Updated pre-commit to a more recent one:

1. added trivy instead of tfsec
2. added tflint
3. added terraform provider command to enable lock for different machines

### Major Changes

`static_analysis.yaml` now call the dx template.
pre-commit work as before, but tfsec was deprecated in favor of trivy and now there is also tflint for another check.
For the correct work of the new static analysis action there is a new pre commit hook that execute the command
```
  terraform providers lock \
    -platform=windows_amd64 \
    -platform=darwin_amd64 \
    -platform=darwin_arm64 \
    -platform=linux_amd64
```
for each `.terraform.lock.hcl` file with a change.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -42,3 +42,6 @@ __TMP
 .metals/
 __azurite_*
 /.idea
+
```
Comment (Krusty93) [inline .gitignore:45]: Can be removed?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
     hooks:
+      - id: terraform_tflint
+        args:
+          - --args=--disable-rule terraform_required_version
+          - --args=--config=__GIT_WORKING_DIR__/.tflint.hcl
       - id: terraform_fmt
       - id: terraform_docs
+        name: terraform_docs on modules
         args:
+          # - --hook-config=--create-file-if-not-exist=true
```
Comment (Krusty93) [inline .pre-commit-config.yaml:22]: Why is this removed?

File: .github/workflows/static_analysis.yml [hunk 3]
```diff
@@ -14,93 +14,10 @@ on:
...
-
-      - name: run_pre_commit_terraform
-        run: |
-          TAG="v1.96.1@sha256:9aea677ac51d67eb96b3bbb4cf93b16afdde5476f984e75e87888850d18146c9"
-          docker run -v $(pwd):/lint -w /lint ghcr.io/antonbabenko/pre-commit-terraform:$TAG run -a
+    uses:  pagopa/dx/.github/workflows/static_analysis.yaml@main
+    name: Terraform Validation
+    secrets: inherit
+    with:
+      terraform_version: "1.7.5"
```
Comment (christian-calabrese) [inline .github/workflows/static_analysis.yml:21]: Do you think we can get the terraform version from the .terraform-version file?

File: .utils/terraform_lock_precommit.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  # Run terraform providers lock
+  terraform providers lock \
+    -platform=windows_amd64 \
+    -platform=darwin_amd64 \
+    -platform=darwin_arm64 \
+    -platform=linux_amd64
+
+  # Return to the previous directory
+  cd - > /dev/null
+done
```
Comment (christian-calabrese) [inline .utils/terraform_lock_precommit.sh:29]: Can we move this script in dx and use it as a pre-commit in the other repos?

Comment (christian-calabrese) [reply .utils/terraform_lock_precommit.sh:29]: Pretty much as it's been done for the script that checks the modules lock file
https://github.com/pagopa/dx/blob/main/.pre-commit-hooks.yaml

File: .github/workflows/static_analysis.yml (hunk 3 above)
Comment (mamu0) [reply .github/workflows/static_analysis.yml:21]: I think that is tricky, in a github action if you are calling a workflow you can't read a file content

File: .pre-commit-config.yaml (hunk 2 above)
Comment (mamu0) [reply .pre-commit-config.yaml:22]: It’s commented out because it wasn’t initially included. Many resources currently lack a `README.md` file, and I wanted to discuss whether we should create them. 
To maintain consistency, I believe we should automate their creation.

File: .gitignore (hunk 1 above)
Comment (mamu0) [reply .gitignore:45]: I wanted to bring this up as well. By default, all .terraform.lock.hcl files are being pushed, but do we want it that way? This would include files generated locally in `modules`, such as those in `common`.

File: .utils/terraform_lock_precommit.sh (hunk 4 above)
Comment (mamu0) [reply .utils/terraform_lock_precommit.sh:29]: this is a great idea! I'll try to implement it

Comment (mamu0) [reply .utils/terraform_lock_precommit.sh]: done!

**Review by christian-calabrese — APPROVED** [review]
